### ISSUE TITLE
Fallback serializer in contextual serializer is missing on Kotlin 1.8.20 

### ISSUE THREAD

**Issue by sandwwraith** [Kotlin/kotlinx.serialization#2158]
**Describe the bug**
```kotlin
    @Serializable
    data class Response<T>(
        val ok: Boolean,
        @Contextual
        val result: T?
    )

    @Test
    fun testFallback() {
        val serializer = serializer<Response<List<String>>>()
        val resp = Response(true, listOf("a", "b"))
        println(Json.encodeToString(serializer, resp))
    }
```

On 1.8.0 it prints `{"ok":true,"result":["a","b"]}`. On 1.8.20, it results in 

```
kotlinx.serialization.SerializationException: Serializer for class 'Any' is not found.
Mark the class as @Serializable or provide the serializer explicitly.
	at kotlinx.serialization.internal.Platform_commonKt.serializerNotRegistered(Platform.common.kt:91)
	at kotlinx.serialization.ContextualSerializer.serializer(ContextualSerializer.kt:51)
	at kotlinx.serialization.ContextualSerializer.serialize(ContextualSerializer.kt:63)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeSerializableValue(StreamingJsonEncoder.kt:224)
	at kotlinx.serialization.encoding.Encoder$DefaultImpls.encodeNullableSerializableValue(Encoding.kt:299)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeNullableSerializableValue(AbstractEncoder.kt:18)
	at kotlinx.serialization.encoding.AbstractEncoder.encodeNullableSerializableElement(AbstractEncoder.kt:90)
	at kotlinx.serialization.json.internal.StreamingJsonEncoder.encodeNullableSerializableElement(StreamingJsonEncoder.kt:154)
	at Response.write$Self(starProjections.kt:17)
```

Likely because `fallbackSerializer` in https://github.com/Kotlin/kotlinx.serialization/blob/63eff4d055293383fe2783105c3860c214ae8a46/core/commonMain/src/kotlinx/serialization/ContextualSerializer.kt#L51 is not passed by plugin.

**Environment**
 - Kotlin version: 1.8.20-dev-5812
 - Library version: 1.4.1
 - Kotlin platforms: JVM

**Comment by sandwwraith** [comment]
@shanshin It is connected with caching as far as I can see

**Comment by shanshin** [comment]
Fixed in 1.8.20 via [d0f4532...](https://github.com/JetBrains/kotlin/commit/d063db3ce0631c84041c24024d04c134339dc38a)
